### ISSUE TITLE
Improve `run-ios` experience when selecting a device/simulator

### ISSUE THREAD

**Issue by lucasbento** [react-native-community/cli#367]
## Describe the Feature

This is similar to what has been brought up on [#142](https://github.com/react-native-community/cli/issues/142#issuecomment-460011052), basically add a simpler way to specify which device/simulator you want to run your app on.

## Possible Implementations

```sh
react-native run-ios --device="iPhone X" # will pick up the first iPhone X
react-native run-ios --device="iPhone X 12.1"
react-native run-ios --list-devices # will list devices for iOS
react-native --list-devices # list connected devices, maybe for both platforms
```

**Comment by lucasbento** [comment]
cc @satya164, perhaps you have some thoughts on this as you are working on #142.

**Comment by lucasbento** [comment]
Similar work has been done on https://github.com/wswebcreation/start-ios-simulator.

**Comment by thymikee** [comment]
Can you elaborate more on how to make the experience better? We already have `--device` flag and by default we run first device/some simulator (not sure about the logic here yet). It would be bad if we made the `run-ios` command always interactive imo.

**Comment by lucasbento** [comment]
I'm also against making the `run-ios` command always interactive, this is more like an idea to add a different way of selecting a device to run on _and_ to improve the `--device` argument to accept part of device names.

Currently if you specify `--udid` or `--device` wrong you will get an error with a list of available devices which is currently the only way that you can see their info in the CLI.

**Comment by adamTrz** [comment]
Not stale

**Comment by adamTrz** [comment]
This will be fixed by #1676 once that's merged.
